backend/dataall/modules/metadata_forms/services/attached_metadata_form_service.py

```diff
@@ -76,5 +86,13 @@ def list_attached_forms(filter=None):
     @staticmethod
     def delete_attached_metadata_form(uri):
         mf = AttachedMetadataFormService.get_attached_metadata_form(uri)
-        with get_context().db_engine.scoped_session() as session:
+        context = get_context()
+        with context.db_engine.scoped_session() as session:
+            ResourcePolicyService.check_user_resource_permission(
+                session=session,
+                username=context.username,
+                groups=context.groups,
+                resource_uri=mf.entityUri,
+                permission_name=ATTACH_METADATA_FORM,  # attach and delete are the same for now
```

Comment:
same as above - any reason we don't use `    @ResourcePolicyService.has_resource_permission(ATTACH_METADATA_FORM, parent_resource=_FUNC_RETURNING_ENTITY_URI)` and instead explicitly call `check_user_resource_permission`

Reply:
same as above